packages/stream_video_flutter/lib/src/call_screen/call_content/picture_in_picture/android_pip_overlay.dart

```diff
@@ -8,35 +13,106 @@ class AndroidPipOverlay extends StatefulWidget {
     this.customBuilder,
-    this.prioritiseScreenSharingTrack = true,
   });
 
   final Call call;
-  final Comparator<CallParticipantState>? sort;
+  final PictureInPictureConfiguration? pictureInPictureConfiguration;
+  @Deprecated('Use [pictureInPictureConfiguration.sort] instead')
+  final Sort<CallParticipantState>? sort;
+  @Deprecated(
+    'Use [pictureInPictureConfiguration.androidPiPConfiguration.customBuilder] instead',
+  )
   final CallWidgetBuilder? customBuilder;
-  final bool prioritiseScreenSharingTrack;
 
   @override
   State<AndroidPipOverlay> createState() => _AndroidPipOverlayState();
 }
 
-class _AndroidPipOverlayState extends State<AndroidPipOverlay> {
+class _AndroidPipOverlayState extends State<AndroidPipOverlay>
+    with CallParticipantsSortingMixin {
+  StreamSubscription<List<CallParticipantState>?>? _participantsSubscription;
+
+  @override
+  Filter<CallParticipantState>? get participantFilter => null;
+
+  @override
+  Sort<CallParticipantState>? get participantSort =>
+      widget.pictureInPictureConfiguration?.sort ?? widget.sort;
+
+  @override
+  void initState() {
+    super.initState();
+    recalculateParticipants(widget.call.state.value.callParticipants);
+
+    _participantsSubscription = widget.call
+        .partialState((state) => state.callParticipants)
+        .listen(recalculateParticipants);
+  }
+
+  @override
+  void dispose() {
+    _participantsSubscription?.cancel();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
+    Widget pipBody = const SizedBox.shrink();
+
+    if (sortedParticipants.isNotEmpty) {
+      final pipParticipant = sortedParticipants.first;
+
+      final hasScreenShare =
+          pipParticipant.isScreenShareEnabled &&
+          pipParticipant.screenShareTrack != null;
+
+      // Show screen share if:
+      // 1. prioritise is true and screen share is available, OR
+      // 2. video is disabled but screen share is available (fallback)
+      final shouldShowScreenShare =
+          hasScreenShare &&
+          (widget.pictureInPictureConfiguration?.pipTrackPriority !=
+                  PipTrackPriority.camera ||
+              !pipParticipant.isVideoEnabled);
+
+      if (shouldShowScreenShare) {
+        pipBody = ScreenShareContent(
+          key: ValueKey(
+            '${pipParticipant.uniqueParticipantKey} - screenShareContent',
+          ),
+          call: widget.call,
+          participant: pipParticipant,
+        );
+      } else {
+        pipBody = StreamCallParticipant(
+          // We use the sessionId as the key to map the state to the participant.
+          key: Key(pipParticipant.uniqueParticipantKey),
+          call: widget.call,
+          participant: pipParticipant,
+        );
+      }
+    }
+
+    final customBuilder =
+        widget
+            .pictureInPictureConfiguration
+            ?.androidPiPConfiguration
+            .callPictureInPictureWidgetBuilder ??
+        widget.customBuilder;
```

Comment:
Not sure if this matters a lot, but if you move this upwards you can do
```dart
if (customBuilder != null && sortedParticipants.isNotEmpty) {
```
because if you have a custom builder you don't use the pipBody.